clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -1076,4 +1077,15 @@ boolean isOutOfMemory() {
     boolean isMadeReadProgressLastPoll() {
         return madeReadProgressLastPoll;
     }
+
+    //package-private for testing
+    boolean isEmpty() {
+        // Split the checks to keep checkstyle happy
+        boolean isEmpty = channels.isEmpty() && closingChannels.isEmpty() &&
+                explicitlyMutedChannels.isEmpty() && stagedReceives.isEmpty();
+        isEmpty &= immediatelyConnectedKeys.isEmpty() && keysWithBufferedRead.isEmpty() &&
+                connected.isEmpty() && disconnected.isEmpty() && failedSends.isEmpty();
+        isEmpty &= completedSends.isEmpty() && completedReceives.isEmpty();
+        return isEmpty;
```

Comment:
My concern with this approach is that we are as likely to forget to update this as we are to forget to update `close`. I wonder if a test that reflectively iterated over all the fields and checked if they were empty would achieve the goal better. What do you think?

Reply:
Done.